packages/state-transition/src/cache/stateCache.ts

```diff
@@ -207,3 +215,13 @@ export function isStateValidatorsNodesPopulated(state: CachedBeaconStateAllForks
 export function isStateBalancesNodesPopulated(state: CachedBeaconStateAllForks): boolean {
   return state.balances["nodesPopulated"] === true;
 }
+
+export function getPubkeyForAggregation(state: CachedBeaconStateAllForks, index: ValidatorIndex): PublicKey {
```

Comment:
You can attach this functions to the epochCache class and take the same arguments